src/contracts/core/AVSDirectory.sol

```diff
@@ -197,44 +197,33 @@ contract AVSDirectory is
     }
 
     /**
-     *  @notice Called by AVSs or operators to remove an operator from an operator set.
+     * @notice Called by an operator to deregister from an operator set
+     * 
+     * @param avs The address of the AVS to deregister the operator from.
+     * @param operatorSetIds The IDs of the operator sets.
+     * 
+     * @dev msg.sender used is the operator
+     */
+    function deregisterFromAVSOperatorSets(
```

Comment:
Based on naming discussions going to put some options here:

```
function deregisterSelfFromAVSOperatorSets(
    address avs, 
    uint32[] calldata operatorSetIds
) external;
``` 

```
function leaveAVSOperatorSets(address avs, uint32[] calldata operatorSetIds) external
```

Reply:
I like the first one :)